src/pymatgen/util/num.py

```diff
@@ -22,14 +30,21 @@ def round_to_sigfigs(num, sig_figs):
     return round(num, prec)
 
 
-def make_symmetric_matrix_from_upper_tri(val):
-    """Given a symmetric matrix in upper triangular matrix form as flat array indexes as:
-    [A_xx,A_yy,A_zz,A_xy,A_xz,A_yz]
+def make_symmetric_matrix_from_upper_tri(val: ArrayLike) -> NDArray:
```

Comment:
Looks like this helpful function is designed to work for 3x3 matrix alone (rank-2 tensor), and wouldn't scale well as the `idx` is hard-coded, also it doesn't work along the diagonals, I would add a check to reject non 3x3 arrays